### ISSUE TITLE
SUDO passwords with non-ascii characters break Ansible

### ISSUE THREAD

**Issue by dagwieers** [ansible/ansible#18029]
##### ISSUE TYPE

- Bug Report
##### COMPONENT NAME

exec_command (local)
##### ANSIBLE VERSION

```
ansible 2.1.0.0
  config file = /home/USER/git/ansible/ansible.cfg
  configured module search path = Default w/o overrides
```
##### SUMMARY

We noticed that using a SUDO password with a non-ascii character causes Ansible to throw an exception when feeding the password to sudo. In my case the password include the ° character (0xb0), like the error output indicates:

```
UnicodeEncodeError: 'ascii' codec can't encode character u'\xb0' in position 0: ordinal not in range(128)
```
##### STEPS TO REPRODUCE

I intend to write a minimal reproducer an using the master branch.
##### ACTUAL RESULTS

```
[USER@SYSTEM ansible]$ ansible-playbook some_playbook.yml -K -vvv
Using /home/USER/git/ansible/ansible.cfg as config file
SUDO password:

PLAY [system01] ****************************************************************

TASK [fetch from system] *******************************************************
task path: /home/USER/git/ansible/some_playbook.yml:32
<localhost> ESTABLISH LOCAL CONNECTION FOR USER: USER
<localhost> EXEC /bin/sh -c '( umask 77 && mkdir -p "` echo /tmp/ansible-tmp-1476453804.28-165187216012413 `" && echo ansible-tmp-1476453804.28-165187216012413="` echo /tmp/ansible-tmp-1476453804.28-165187216012413 `" ) && sleep 0'
<localhost> PUT /tmp/tmpWVAqrj TO /tmp/ansible-tmp-1476453804.28-165187216012413/command
<localhost> EXEC /bin/sh -c 'chown -R sysauto /tmp/ansible-tmp-1476453804.28-165187216012413/ && sleep 0'
<localhost> EXEC /bin/sh -c 'setfacl -R -m u:sysauto:rX /tmp/ansible-tmp-1476453804.28-165187216012413/ && sleep 0'
<localhost> EXEC /bin/sh -c 'sudo -H -S  -p "[sudo via ansible, key=opbxnljylhudewlchnsbwfnibhzyaent] password: " -u sysauto /bin/sh -c '"'"'echo BECOME-SUCCESS-opbxnljylhudewlchnsbwfnibhzyaent; LANG=en_US.UTF-8 LC_ALL=en_US.UTF-8 LC_MESSAGES=en_US.UTF-8 /usr/bin/python /tmp/ansible-tmp-1476453804.28-165187216012413/command'"'"' && sleep 0'
An exception occurred during task execution. The full traceback is:
Traceback (most recent call last):
  File "/usr/lib/python2.7/site-packages/ansible/executor/task_executor.py", line 124, in run
    res = self._execute()
  File "/usr/lib/python2.7/site-packages/ansible/executor/task_executor.py", line 446, in _execute
    result = self._handler.run(task_vars=variables)
  File "/usr/lib/python2.7/site-packages/ansible/plugins/action/normal.py", line 33, in run
    results = merge_hash(results, self._execute_module(tmp=tmp, task_vars=task_vars))
  File "/usr/lib/python2.7/site-packages/ansible/plugins/action/__init__.py", line 644, in _execute_module
    res = self._low_level_execute_command(cmd, sudoable=sudoable, in_data=in_data)
  File "/usr/lib/python2.7/site-packages/ansible/plugins/action/__init__.py", line 718, in _low_level_execute_command
    rc, stdout, stderr = self._connection.exec_command(cmd, in_data=in_data, sudoable=sudoable)
  File "/usr/lib/python2.7/site-packages/ansible/plugins/connection/local.py", line 109, in exec_command
    p.stdin.write(self._play_context.become_pass + '\n')
UnicodeEncodeError: 'ascii' codec can't encode character u'\xb0' in position 0: ordinal not in range(128)

fatal: [system01]: FAILED! => {"failed": true, "msg": "Unexpected failure during module execution.", "stdout": ""}

NO MORE HOSTS LEFT *************************************************************

PLAY RECAP *********************************************************************
system01                   : ok=0    changed=0    unreachable=0    failed=
```

**Comment by dagwieers** [comment]
This seems to be a duplicate of #16557

**Comment by abadger** [comment]
Different connection plugin so it's a different bug.  I have a potential fix for devel: https://github.com/ansible/ansible/pull/18044  Once it passes tests I'll see how far back I can easily backport it (definitely 2.2, not yet sure about 2.1)

**Comment by abadger** [comment]
Fix to the local connection plugin merged to stable-2.1, stable-2.2, and devel (should be in 2.1.3 and 2.2.0)
